2.6 (#206)
nf-core/sarek: docs/usage.md

The pipeline also dynamically loads configurations from [https://github.com/nf-core/configs](https://github.com/nf-core/configs) when it runs, making multiple config profiles for various institutional clusters available at run time. For more information and to see if your system is available in these configs please see the [nf-core/configs documentation](https://github.com/nf-core/configs#documentation).

Note that multiple profiles can be loaded, for example: `-profile test,docker` - the order of arguments is important!
They are loaded in sequence, so later profiles can overwrite earlier profiles.

If `-profile` is not specified, the pipeline will run locally and expect all software to be installed and available on the `PATH`. This is _not_ recommended.

- `docker`
  - A generic configuration profile to be used with [Docker](http://docker.com/)
  - Pulls software from dockerhub: [`nfcore/sarek`](http://hub.docker.com/r/nfcore/sarek/)
- `singularity`
  - A generic configuration profile to be used with [Singularity](https://sylabs.io/docs/)
  - Pulls software from DockerHub: [`nfcore/sarek`](http://hub.docker.com/r/nfcore/sarek/)
- `conda`

>   - Please only use Conda as a last resort `i.e.` when it's not possible to run the pipeline with Docker or Singularity.

Reviewer: ```suggestion
  - Please only use Conda as a last resort i.e. when it's not possible to run the pipeline with Docker or Singularity.
```

  - Includes links to test data so needs no other parameters
- `test_annotation`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters
- `test_no_gatk_spark`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters
- `test_split_fastq`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters
- `test_targeted`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters
- `test_tool`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters
- `test_trimming`
  - A profile with a complete configuration for automated testing
  - Includes links to test data so needs no other parameters

### --input

Use this to specify the location of your input TSV file. For example:
TSV file should correspond to the correct step, see [`--step`](#--step) and [input](input.md) documentation for more information

```bash
--input sample.tsv
```

> Multiple TSV files can be specified if the path must be enclosed in quotes.

Reviewer: Might be worth to specify with wildcard or comma separated list.

> Use this to specify the location to a directory with fastq files on `mapping` step for a single germline sample only.

Reviewer: I don't know what you mean by '**on** mapping step' here

Author: That it works only on the `mapping` step

Reviewer: Ah ok, the nI would say:
```suggestion
Use this to specify the location to a directory with fastq files for the `mapping` step of single germline samples only.
```

For example:

```bash

> --input /path/to/directory

Reviewer: ```suggestion
--input </path/to/directory>
```

Check this throughout the document for consistency with the rest of the docs

Author: Good point

Multiple VCF files can be specified if the path must be enclosed in quotes

### --step

Use this to specify the starting step:
Default `mapping`
Available: `mapping`, `prepare_recalibration`, `recalibrate`, `variant_calling`, `annotate`, `Control-FREEC`

### --help

Will display the help message

### --no_intervals

Disable usage of intervals file, and disable automatic generation of intervals file when none are provided.

### --nucleotides_per_second

Use this to estimate of how many seconds it will take to call variants on any interval, the default value is `1000` is it's not specified in the `<intervals>.bed` file.

### --sentieon

If [Sentieon](https://www.sentieon.com/) is available, use this to enable it for preprocessing, and variant calling.
Adds the following tools for the [`--tools`](#--tools) options: `DNAseq`, `DNAscope` and `TNscope`.

More information in the [sentieon](sentieon.md) documentation.

### --skip_qc

Use this to disable specific QC and Reporting tools.

> Available: `all`, `bamQC`, `BaseRecalibrator`, `BCFtools`, `Documentation`, `FastQC`, `MultiQC`, `samtools`, `vcftools`, `versions`

Reviewer: Can you supply multiple? Or just a single one at a time?

Author: Multiple separated by commas

Reviewer: I would then state that, beecause at the moment that seems to be the only (exclusive) options, as you have `all` as one option..

Default: `None`

### --target_bed

Use this to specify the target BED file for targeted or whole exome sequencing.

### --tools

Use this parameter to specify the variant calling and annotation tools to be used.
Multiple tools can be specified.
For example:

```bash
--tools 'Strelka,mutect2,SnpEff'
```

Available variant callers: `ASCAT`, `ControlFREEC`, `FreeBayes`, `HaplotypeCaller`, `Manta`, `mpileup`, `MSIsensor`, `Mutect2`, `Strelka`, `TIDDIT`.

> > `/!\` Not all variant callers are available for both germline and somatic variant calling.

Reviewer: See previous comment about `/!\` not being clear, **Warning** would better catch the eye

For more details please check the [variant calling](variant_calling.md) extra documentation.

Available annotation tools: `VEP`, `SnpEff`, `merge`. For more details, please check the [annotation](annotation.md) extra documentation.

## Modify fastqs (trim/split)

### --trim_fastq

> Use this to perform adapter trimming [Trim Galore](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md)

Reviewer: ```suggestion
Use this to perform adapter trimming with [Trim Galore](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md)
```

### --clip_r1

Instructs Trim Galore to remove a number of bp from the 5' end of read 1 (or single-end reads).
This may be useful if the qualities were very poor, or if there is some sort of unwanted bias at the 5' end.

### --clip_r2

Instructs Trim Galore to remove a number of bp from the 5' end of read 2 (paired-end reads only).
This may be useful if the qualities were very poor, or if there is some sort of unwanted bias at the 5' end.

### --three_prime_clip_r1

Instructs Trim Galore to remove a number of bp from the 3' end of read 1 (or single-end reads) AFTER adapter/quality trimming has been performed.
This may remove some unwanted bias from the 3' end that is not directly related to adapter sequence or basecall quality.

### --three_prime_clip_r2

Instructs Trim Galore to remove a number of bp from the 3' end of read 2 AFTER adapter/quality trimming has been performed.
This may remove some unwanted bias from the 3' end that is not directly related to adapter sequence or basecall quality.

### --trim_nextseq

> This enables the option `--nextseq-trim=3'CUTOFF` within `Cutadapt`, which will set a quality cutoff (that is normally given with `-q` instead), but qualities of G bases are ignored.

Reviewer: As an outsider/non-user, I thought you were using `TrimGalore` rather than `CutAdapt`? Or is this for another purpose?

I can only _now_ see reading desciprtion this is for poly-G trimming; so I would maybe make the parameter clearer/more informative (same as `--trim_fastq`, polyG trimming also technically trimming)

Author: No idea.
It was a PR made during the hackathon, and I think that these params and docs were copied from another nf-core pipeline

Reviewer: Ok. In `--help` it also says its TrimGalore, so I guess that's leftover.

This trimming is in common for the NextSeq and NovaSeq-platforms, where basecalls without any signal are called as high-quality G bases.

### --save_trimmed

Option to keep trimmed FASTQs

### --split_fastq

> 

Reviewer: Maybe mention of brief purpose of this, even though you've given a link.

Author: I think that the name and the description is enough.
Not sure that many people actually use that, but could be useful if you want to speed things up, or if you have a really big fastq as an input

--split_fastq 10000
```

## Preprocessing

### --markdup_java_options

To control the java options necessary for the GATK `MarkDuplicates` process, you can set this parameter.
Default: "-Xms4000m -Xmx7g"
For example:

```bash
--markdup_java_options "-Xms4000m -Xmx7g"
```

### --no_gatk_spark

Use this to disable usage of GATK Spark implementation of their tools in local mode.

### --save_bam_mapped

Will save mapped BAMs.

### --skip_markduplicates

Will skip MarkDuplicates. This params will also save the mapped BAMS, to enable restart from step `prepare_recalibration`

## Variant Calling

### --ascat_ploidy

Use this parameter to overwrite default behavior from ASCAT regarding ploidy.
Requires that [`--ascat_purity`](#--ascat_purity) is set

### --ascat_purity

Use this parameter to overwrite default behavior from ASCAT regarding purity.
Requires that [`--ascat_ploidy`](#--ascat_ploidy) is set

### --cf_coeff

Control-FREEC `coefficientOfVariation`
Default: 0.015

### --cf_ploidy

Control-FREEC `ploidy`
Default: 2

### --cf_window

Control-FREEC `window size`
Default: Disabled

### --no_gvcf

Use this to disable g.vcf output from `GATK HaplotypeCaller`.

### --no_strelka_bp

> Use this not to use `Manta` `candidateSmallIndels` for `Strelka` (not recommended by Best Practices).

Reviewer: ```suggestion
Use this not to use `Manta` `candidateSmallIndels` for `Strelka` (not recommended by Broad Institute's Best Practices).
```

It is recommended to make your own panel-of-normals, as it depends on sequencer and library preparation.
For tests in iGenomes there is a dummy PON file in the Annotation/GermlineResource directory, but it _should not be used_ as a real panel-of-normals file.
Provide your PON by:

```bash
--pon '[path to the PON VCF]'
```

PON file should be bgzipped.

### --pon_index

Tabix index of the panel-of-normals bgzipped VCF file.
If none provided, will be generated automatically from the panel-of-normals bgzipped VCF file.

## Annotation

### --annotate_tools

Specify from which tools Sarek should look for VCF files to annotate, only for step `Annotate`.
Available: `HaplotypeCaller`, `Manta`, `Mutect2`, `Strelka`, `TIDDIT`
Default: `None`

### --annotation_cache

Enable usage of annotation cache, and disable usage of already built containers within Sarek.
For more information, follow the [annotation guidelines](annotation.md#using-downloaded-cache).

```bash
--annotation_cache
```

### --snpeff_cache

To be used conjointly with [`--annotation_cache`](#--annotation_cache), specify the cache snpEff directory:

```bash

> --snpeff_cache /path/to/snpeff_cache

Reviewer: ```suggestion
--snpeff_cache </path/to/snpeff_cache>
```

```

### --vep_cache

To be used conjointly with [`--annotation_cache`](#--annotation_cache), specify the cache VEP directory:

```bash

> --vep_cache /path/to/vep_cache

Reviewer: ```suggestion
--vep_cache </path/to/vep_cache>
```

Specify base path to AWS iGenomes
Default: `s3://ngi-igenomes/igenomes/`

### --igenomes_ignore

Do not load `igenomes.config` when running the pipeline.
You may choose this option if you observe clashes between custom parameters and those supplied in `igenomes.config`.
This option will load the `genomes.config` file instead.
You can then specify the `--genome custom` and specify any reference file on the command line or within a config file.

```bash
--igenomes_ignore
```

### --genomes_base

Specify base path to reference genome

### --save_reference

Enable saving reference indexes and other files built within Sarek.

```bash
--save_reference
```

### --ac_loci

If you prefer, you can specify the full path to your reference genome when you run the pipeline:

```bash
--ac_loci '[path to the loci file]'
```

### --ac_loci_gc

If you prefer, you can specify the full path to your reference genome when you run the pipeline:

```bash
--ac_loci_gc '[path to the loci GC file]'
```

### --bwa

If you prefer, you can specify the full path to your reference genome when you run the pipeline:

> If none provided, will be generated automatically from the fasta reference.

```bash

> --bwa '[path to the BWA indexes]'

Reviewer: This might be a bit confusing because there are multiple BWA indexes depending on which algo. you use. Maybe give more explicit example?

Reviewer: For all index path examples, you should follow your </path/to/XXX> convention from the other docs

Author: Good point, I'll add that

Use to set a top-limit for the default CPU requirement for each process.
Should be a string in the format integer-unit eg. `--max_cpus 1`

### --single_cpu_mem

Use to set memory for a single CPU.
Should be a string in the format integer-unit eg. `--single_cpu_mem '8.GB'`

## AWSBatch specific parameters

Running the pipeline on AWSBatch requires a couple of specific parameters to be set according to your AWSBatch configuration.
Please use [`-profile awsbatch`](https://github.com/nf-core/configs/blob/master/conf/awsbatch.config) and then specify all of the following parameters.

### --awsqueue

The JobQueue that you intend to use on AWSBatch.

### --awsregion

The AWS region to run your job in.
Default is set to `eu-west-1` but can be adjusted to your needs.

### --awscli

The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installation) path in your custom AMI.
Default: `/home/ec2-user/miniconda/bin/aws`.

Please make sure to also set the `-w/--work-dir` and `--outdir` parameters to a S3 storage bucket of your choice - you'll get an error message notifying you if you didn't.

## Deprecated params

> > :warning: These params are deprecated -- They will be removed in a future release.

Reviewer: You've removed `:warning:` elsewhere, should do that here too - but check my commetns elsewhere.
